package messages;

message VisionBall {
	optional float distance = 1;
        optional float bearing = 2;
        optional float bearing_deg = 3;
        optional float angle_x_deg = 4;
        optional float angle_y_deg = 5;
        optional float elevation_deg = 6;
        optional float distance_sd = 7;
        optional float bearing_sd = 8;
	optional bool inTopCam = 9;
	optional float radius = 10;
	optional int32 confidence = 11;
	optional float heat = 12;
	optional bool on = 13;
	optional int32 frames_on = 14;
	optional int32 frames_off = 15;
}

message FilteredBall
{
	optional VisionBall vis = 1;
	optional float distance = 2;
	optional float bearing = 3;
        optional float bearing_deg = 4;
        optional float x = 5;
        optional float y = 6;
	optional float rel_x = 7;
	optional float rel_y = 8;
}